actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -37,6 +37,14 @@ def cache_collection_render(instrumentation_payload, view, template, collection)
         # Extract the items for the keys that are not found
         collection = keyed_collection.reject { |key, _| cached_partials.key?(key) }.values
 
+        # If new entries are available to the collection and cache invalidation on
+        # new entry is required, we clear the cache and set the collection as the
+        # initial one
+        if !collection.empty? && @options[:cached].is_a?(Hash) && @options[:cached][:invalidate_on_new_entry]
+          collection = keyed_collection.values
+          cached_partials = {}
+        end
+
```

Comment:
I guess a hash value is still truthy so the original behavior of `cached: true` Just Works? Is there any assumptions that this _has_ to be a boolean? 🤔

Reply:
> I guess a hash value is still truthy so the original behavior of cached: true Just Works?

The only references I found, in `#will_cache?` exepct a truthy value and not a boolean. 

```ruby
# L16 

def will_cache?(options, view)
    options[:cached] && view.controller.respond_to?(:perform_caching) && view.controller.perform_caching
  end
```

However I just realized that a callable object can be passed to `cached:`, to allow for custom cache keys. 

To have cache invalidation working for callable cache keys, I'd need to also pass the Proc to the options Hash, and update `#callable_cache_key?` to take this into account : 

```ruby 
<%= render partial: "pictures/picture", collection: @pictures, cached: { invalidate_on_new_entry: true, cahe_key_proc: -> picture { "cached_picture" }  }  %> 

def callable_cache_key?
    @options[:cached].respond_to?(:call) ||
      (@options[:cached].is_a?(Hash) && @options[:cached][:cache_key_proc] && @options[:cached][:cache_key_proc].respond_to?(:call))
  end
```